test/com/amazon/ion/impl/IonReaderContinuableTopLevelBinaryTest.java

```diff
@@ -1813,6 +1856,995 @@ public void symbolTableSnapshotImplementsBasicMethods() throws Exception {
+            "\"abc\" " +
+            "\"abcdefghijklmnopqrstuvwxyz\" " +
+            "\"def\""
+        );
+
+        IonReader reader = boundedReaderFor(new ByteArrayInputStream(bytes), 8, 16, byteAndOversizedValueCountingHandler);
+
+        assertEquals(IonType.STRING, reader.next());
+        assertEquals("abc", reader.stringValue());
+        assertEquals(1, oversizedCounter.get());
+        assertEquals(IonType.STRING, reader.next());
+        assertEquals("def", reader.stringValue());
+        assertNull(reader.next());
+        reader.close();
+        assertEquals(2, oversizedCounter.get());
+        totalBytesInStream = bytes.length;
+        assertBytesConsumed();
+    }
+
+    @Test
+    public void oversizeValueDetectedDuringScalarFillIncremental() throws Exception {
+        byte[] bytes = toBinary(
+            "\"abcdefghijklmnopqrstuvwxyz\" " + // Requires 32 bytes (4 IVM, 1 TID, 1 length, 26 chars)
+            "\"abc\" " +
+            "\"abcdefghijklmnopqrstuvwxyz\" " +
+            "\"def\""
+        );
+
+        ResizingPipedInputStream pipe = new ResizingPipedInputStream(bytes.length);
+        IonReader reader = boundedReaderFor(pipe, 8, 16, byteAndOversizedValueCountingHandler);
+        int valueCounter = 0;
+        for (byte b : bytes) {
+            pipe.receive(b);
+            IonType type = reader.next();
+            if (type != null) {
+                valueCounter++;
+                assertTrue(valueCounter < 3);
+                if (valueCounter == 1) {
+                    assertEquals(IonType.STRING, type);
+                    assertEquals("abc", reader.stringValue());
+                    assertEquals(1, oversizedCounter.get());
+                } else {
+                    assertEquals(2, valueCounter);
+                    assertEquals(IonType.STRING, type);
+                    assertEquals("def", reader.stringValue());
+                    assertEquals(2, oversizedCounter.get());
+                }
+            }
+        }
+        assertEquals(2, valueCounter);
+        assertNull(reader.next());
+        reader.close();
+        assertEquals(2, oversizedCounter.get());
+        totalBytesInStream = bytes.length;
+        assertBytesConsumed();
+    }
+
+    @Test
+    public void oversizeValueDetectedDuringFillOfOnlyScalarInStream() throws Exception {
+        // Unlike the previous test, where excessive size is detected when trying to skip past the value portion,
+        // this test verifies that excessive size can be detected while reading a value header, which happens
+        // byte-by-byte.
+        byte[] bytes = toBinary("\"abcdefghijklmnopqrstuvwxyz\""); // Requires a 2-byte header.
+        IonReader reader = boundedReaderFor(new ByteArrayInputStream(bytes), 5, 5, byteAndOversizedValueCountingHandler);
+
+        // The maximum buffer size is 5, which will be exceeded after the IVM (4 bytes), the type ID (1 byte), and
```

Comment:
I don't think holding 4 bytes for the IVM is a big deal, but I'm curious about why...

Reply:
The comment is misleading, so I've updated it. The fact that the value is determined to be oversized doesn't actually have to do with the IVM, as the four IVM bytes are freed from the buffer after they are processed. The value is oversized because its header declares a length of 26, but the buffer can only hold 5 bytes.